src/NuGet.Core/NuGet.Packaging/Rules/UpholdBuildConventionRule.cs

```diff
@@ -103,11 +144,44 @@ internal class ConventionViolator
 
             public ConventionViolator(string filePath, string extension, string expectedFile)
             {
-                Path = filePath.Replace(filePath.Split('/')[filePath.Count(p => p == '/')], string.Empty);
+#if DEBUG
```

Comment:
These seems to be good validations to be in product code. Can we remove DEBUG if-region?